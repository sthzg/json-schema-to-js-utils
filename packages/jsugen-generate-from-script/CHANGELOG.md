# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.14.3](https://github.com/sthzg/jsugen/compare/v0.14.2...v0.14.3) (2019-06-25)

**Note:** Version bump only for package @sthzg/jsugen-generate-from-script





## [0.14.2](https://github.com/sthzg/jsugen/compare/v0.14.1...v0.14.2) (2019-06-24)

**Note:** Version bump only for package @sthzg/jsugen-generate-from-script





## [0.14.1](https://github.com/sthzg/jsugen/compare/v0.14.0...v0.14.1) (2019-06-23)

**Note:** Version bump only for package @sthzg/jsugen-generate-from-script





# [0.14.0](https://github.com/sthzg/jsugen/compare/v0.13.0...v0.14.0) (2019-06-22)

### Bug Fixes

- Refactor sources to core to prevent cyclic dependencies ([bab8fe0](https://github.com/sthzg/jsugen/commit/bab8fe0))

### Features

- Add first iteration of from-script generator ([b6564ef](https://github.com/sthzg/jsugen/commit/b6564ef))
