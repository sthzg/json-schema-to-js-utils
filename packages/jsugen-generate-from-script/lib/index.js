import { map, tap, concatMap } from 'rxjs/operators';
import {
  DEFAULT_FILE_DOCSTRING,
  DEFAULT_PRETTIER_OPTIONS,
  withPrependToString,
  withPrettier,
  withWrite,
} from '@sthzg/jsugen-core';
import { fromSourceFile } from '@sthzg/jsugen-core/lib/sources/sourceFile';
import { executeMain } from './internals';

export function generate({ sourceFile, writeConfig }) {
  // ---
  // Configure Transformer Factories.
  // ---
  const prependHeaders = withPrependToString(DEFAULT_FILE_DOCSTRING);
  const prettify = withPrettier(DEFAULT_PRETTIER_OPTIONS);
  const write = withWrite(writeConfig);

  // ---
  // Observable.
  // ---
  return fromSourceFile(sourceFile).pipe(
    concatMap(executeMain),
    map(prependHeaders),
    map(prettify),

    /* Output */
    tap(write),
  );
}
