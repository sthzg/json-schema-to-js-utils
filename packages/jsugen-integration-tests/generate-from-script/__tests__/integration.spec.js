import path from 'path';
import { generate } from '@sthzg/jsugen-generate-from-script';
import {
  DryRunWriteConfig as writeConfig,
  TEST_RESOURCES_DIR,
} from '../../testUtils/helpers';

const TEST_FILE = path.join(
  TEST_RESOURCES_DIR,
  'fromScript/example.codegen.js',
);

describe('generate-from-script', () => {
  test('should pass integration test', done => {
    expect.assertions(1);

    generate({ sourceFile: TEST_FILE, writeConfig }).subscribe(output => {
      expect(output).toMatchSnapshot();
      done();
    });
  });
});
