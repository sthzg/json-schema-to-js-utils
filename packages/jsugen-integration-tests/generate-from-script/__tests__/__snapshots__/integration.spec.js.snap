// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-from-script should pass integration test 1`] = `
"/**
 * @file This file is auto-generated. Do not modify its contents manually.
 */

export const A = 'A';
export const B = A;
"
`;
